Miscellaneous documentation and meta file changes

martinwe-adfinis opened the pull request:
Notably:

 * Add a contribution guideline. It's fairly rudimentary for now, but it should cover the most important bits about the release process.
 * Add @aconitumnapellus to the CODEOWNERS for the GitHub Actions part (consequently, I'm adding you as reviewer as well :slightly_smiling_face:).
 * Remove (what I consider) unrequired collections dependencies from `meta/main.yml`.

Note that due to the now explicitly mentioned support policy (and the rule for old releases in the contribution guidelines), merging this will then also result in me deleting the old release branches.

aconitumnapellus reviewed (approved):
everything LGTM to me

martinwe-adfinis commented:
> Note that due to the now explicitly mentioned support policy (and the rule for old releases in the contribution guidelines), merging this will then also result in me deleting the old release branches.

Did this, and to fully adhere to the contribution guides, I also did a dirty thing:

 * Re-tagged `v0.1` as `v0.1.0`, `v0.2` as `v0.2.0`, and *deleted the old tags* :scream:.
 * Renamed the `v0.3.x` branch to `v0.3`.

Downstream may need to maybe clean up locally (or more easily: just re-fetch the entire repo fresh). Apologies for the inconvenience caused.

(note that this should not apply to `ansible-galaxy`, as AFAICT it does a full clone anyway)
